design/secure-communication.md

```diff
@@ -0,0 +1,94 @@
+# Secure Communication for Contour and Envoy
+
+Status: Draft
+
+This document outlines a design for securing gRPC communication between Contour and Envoy, including all the xDS endpoints.
+
+## Goals
+
+- Achieve secure communication between Contour and Envoy, which will allow a split deployment.
+- Allow use of the SDS service in Contour with authentication.
+
+## Non Goals
+
+- Certificate management, including creation or rotation. These are left as an exercise for the reader at this time.
+
+## Definitions
+
+### mTLS
+
+In this document, we use **mTLS** as shorthand for 'mutual TLS client certificate authentication', without implying a key lifetime. That is, we're just talking about both sides having TLS certs, not implying that we're using the more modern form which implies dynamic, short-lifetime certificates managed by an automated mechanism.
+
+## Background
+
+Currently, the default deployment of Contour includes a Contour and an Envoy container, that connect via the Pod's loopback interface. This uses the pod-level security boundary to provide transport security (no-one can listen in) and authentication (no-one other than Envoy can connect).
```

Comment:
s/includes ..../colocates the Envoy and Contour containers in the same pod communicating over the pod's lookback interface.